lib/elixir/lib/kernel.ex

```diff
@@ -261,6 +261,8 @@ defmodule Kernel do
   @doc """
   Returns the head of a list. Raises `ArgumentError` if the list is empty.
 
+  It works with improper lists.
```

Comment:
Maybe something like "Does not inspect the list to check if it's proper" would be better?